### PR TITLE
Remove circular dependency

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/9277

### CODE REVIEW

File: datafusion/physical-expr/Cargo.toml [hunk 1]
```diff
@@ -74,7 +74,6 @@ uuid = { version = "^1.2", features = ["v4"] }
 
 [dev-dependencies]
 criterion = "0.5"
-datafusion = { workspace = true }
```
Comment (viirya) [inline datafusion/physical-expr/Cargo.toml:77]: Hm, why we added this dependency. Looks not correct. 🤔

**Review by viirya — APPROVED** [review]

File: datafusion/physical-expr/src/regex_expressions.rs [hunk 2]
```diff
@@ -65,7 +65,7 @@ macro_rules! fetch_string_arg {
 ///
 /// # Examples
 ///
-/// ```
+/// ```ignore
```
Comment (viirya) [inline datafusion/physical-expr/src/regex_expressions.rs:68]: 👍

File: datafusion/physical-expr/Cargo.toml (hunk 1 above)
Comment (andygrove) [reply datafusion/physical-expr/Cargo.toml:77]: It looks like it was added to allow for some example code in rustdoc.

**Review by comphead — APPROVED** [review]
Thanks @andygrove are you planning to merge it?

File: datafusion/physical-expr/Cargo.toml (hunk 1 above)
Comment (alamb) [reply datafusion/physical-expr/Cargo.toml:77]: Sorry I missed this on initial review.

**Pull request merged**